app/src/androidMain/kotlin/de/westnordost/streetcomplete/quests/level/LevelUtils.kt

```diff
@@ -0,0 +1,45 @@
+package de.westnordost.streetcomplete.quests.level
+
+import de.westnordost.streetcomplete.data.elementfilter.toElementFilterExpression
+import de.westnordost.streetcomplete.data.osm.geometry.ElementPolygonsGeometry
+import de.westnordost.streetcomplete.data.osm.mapdata.Element
+import de.westnordost.streetcomplete.data.osm.mapdata.MapDataWithGeometry
+import de.westnordost.streetcomplete.util.math.contains
+import de.westnordost.streetcomplete.util.math.isInMultipolygon
+
+val thingsWithLevelFilter by lazy { """
+        nodes, ways, relations with level
+    """.toElementFilterExpression() }
+
+/* only nodes because ways/relations are not likely to be floating around freely in a mall
+ * outline */
+val filter by lazy { """
+        nodes with
+          !level
+          and (name or brand or noname = yes or name:signed = no)
+    """.toElementFilterExpression() }
```

Comment:
well, uh, that would mean that only for things with a name, a quest would be created.

Also, a project-wide public `val` named `filter` is going to be quite confusing.

Anyway, rather than more forth and back, I will just do the remaining review suggestions myself.